packages/react-ui/src/features/flows/components/flow-table.tsx

```diff
@@ -99,28 +62,286 @@ const filters: DataTableFilter[] = [
   },
 ];
 
-async function fetchData(queryParams: URLSearchParams) {
-  return flowsApi.list({
-    projectId: authenticationSession.getProjectId(),
-    cursor: queryParams.get('cursor') ?? undefined,
-    limit: parseInt(queryParams.get('limit') ?? '10'),
-    status: (queryParams.getAll('status') ?? []) as FlowStatus[],
-  });
-}
-
 const FlowsTable = () => {
   const navigate = useNavigate();
+  const [isRenameDialogOpen, setIsRenameDialogOpen] = useState(false);
+  const [isShareDialogOpen, setIsShareDialogOpen] = useState(false);
+  const [isDeleteDialogOpen, setIsDeleteDialogOpen] = useState(false);
+  const [selectedFlow, setSelectedFlow] = useState<{
+    flowId: string;
+    flowName: string;
+    flowVersion?: FlowVersion;
+  }>({
+    flowId: '',
+    flowName: '',
+    flowVersion: undefined,
+  });
+
+  const { refetch } = flowsHooks.useFlowsListing();
+
+  async function fetchData(queryParams: URLSearchParams) {
+    return flowsApi.list({
+      projectId: authenticationSession.getProjectId(),
+      cursor: queryParams.get('cursor') ?? undefined,
+      limit: parseInt(queryParams.get('limit') ?? '10'),
+      status: (queryParams.getAll('status') ?? []) as FlowStatus[],
+    });
+  }
+
+  const { mutate: duplicateFlow } = useMutation({
```

Comment:
Can we extract this to flow-actions-menu for example? because this will be reused inside the buidler let's not duplicate the funcionality maybe seperate the dropdown into flows-action-menu component and use it there?